data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/export/ExportScheduler.java

```diff
@@ -39,10 +51,183 @@ public ExportScheduler(final EnhancedSourceCoordinator sourceCoordinator,
+        } catch (Exception e) {
+            LOG.warn("Check snapshot status for {} failed", snapshotId, e);
+            sourceCoordinator.giveUpPartition(exportPartition);
+            return null;
+        }
+
+        LOG.info("Creating an export task for db {} from snapshot {}", exportPartition.getDbIdentifier(), snapshotId);
+        String exportTaskId = exportTaskManager.startExportTask(
+                snapshotInfo.getSnapshotArn(), progressState.getIamRoleArn(), progressState.getBucket(),
+                progressState.getPrefix(), progressState.getKmsKeyId(), progressState.getTables());
+
+        if (exportTaskId != null) {
+            LOG.info("Export task id is {}", exportTaskId);
+            progressState.setExportTaskId(exportTaskId);
+            sourceCoordinator.saveProgressStateForPartition(exportPartition, null);
+        } else {
+            LOG.error("The export task failed to create, it will be retried");
+            closeExportPartitionWithError(exportPartition);
+            return null;
+        }
+
+        return exportTaskId;
+    }
+
+    private void closeExportPartitionWithError(ExportPartition exportPartition) {
+        ExportProgressState exportProgressState = exportPartition.getProgressState().get();
+        // Clear current task id, so that a new export can be submitted.
+        exportProgressState.setExportTaskId(null);
+        sourceCoordinator.closePartition(exportPartition, DEFAULT_CLOSE_DURATION, DEFAULT_MAX_CLOSE_COUNT);
+    }
+
+    private String checkSnapshotStatus(String snapshotId, Duration timeout) {
+        final Instant startTime = Instant.now();
+        final Instant endTime = startTime.plus(timeout);
+
+        LOG.debug("Start checking status of snapshot {}", snapshotId);
+        while (Instant.now().isBefore(endTime)) {
+            SnapshotInfo snapshotInfo = snapshotManager.checkSnapshotStatus(snapshotId);
+            String status = snapshotInfo.getStatus();
+            // Valid snapshot statuses are: available, copying, creating
+            // The status should never be "copying" here
+            if (SnapshotStatus.AVAILABLE.getStatusName().equals(status)) {
+                LOG.info("Snapshot {} is available.", snapshotId);
+                return status;
+            }
+
+            LOG.debug("Snapshot {} is still creating. Wait and check later", snapshotId);
+            try {
+                Thread.sleep(DEFAULT_CHECK_STATUS_INTERVAL_MILLS);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        throw new RuntimeException("Snapshot status check timed out.");
+    }
+
+    private String checkExportStatus(ExportPartition exportPartition) {
+        long lastCheckpointTime = System.currentTimeMillis();
+        String exportTaskId = exportPartition.getProgressState().get().getExportTaskId();
+
+        LOG.debug("Start checking the status of export " + exportTaskId);
```

Comment:
Use string interpolation here:

```suggestion
        LOG.debug("Start checking the status of export {}", exportTaskId);
```

Reply:
Fixed.